interfaces/seccomp/backend.go

```diff
@@ -0,0 +1,115 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+// Package seccomp implements integration between snappy and
+// ubuntu-core-launcher around seccomp.
+//
+// Snappy creates so-called seccomp profiles for each application (for each
+// snap) present in the system.  Upon each execution of ubuntu-core-launcher,
+// the profile is read and "compiled" to an eBPF program and injected into the
+// kernel for the duration of the execution of the process.
+//
+// There is no binary cache for seccomp, each time the launcher starts an
+// application the profile is parsed and re-compiled.
+//
+// The actual profiles are stored in /var/lib/snappy/seccomp/profiles.
+// This directory is hard-coded in ubuntu-core-launcher.
+package seccomp
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/ubuntu-core/snappy/dirs"
+	"github.com/ubuntu-core/snappy/interfaces"
+	"github.com/ubuntu-core/snappy/osutil"
+	"github.com/ubuntu-core/snappy/snap"
+)
+
+// Backend is responsible for maintaining seccomp profiles for ubuntu-core-launcher.
+type Backend struct {
+}
+
+// Configure creates seccomp security profiles specific to a given snap. The
+// snap can be in developer mode to make security violations non-fatal to the
+// offending application process.
```

Comment:
This isn't true (yet) though I understand the goal is for that to be the case. There is some (light) kernel work and some launcher work to make developer mode work like this. This may end up for GA.

I mention this because it isn't clear how we will flag a seccomp profile as in complain mode. OTOH, adding @complain (ideally) at the top of the file could flag the launcher to use SCMP_ACT_ERRNO or SCMP_ACT_TRACE. Please feel free to add @complain to the top of the profile-- the launcher will currently ignore it but your code here can work.